install_ads

```diff
@@ -657,6 +655,9 @@ done
 for file in "$ADS_DIR/$CONDA_DIR"/envs/venv_ads/bin/*download*; do
   cp "$file" "$ADS_DIR/$BIN_DIR/" || die "Problem creating launchers!"
 done
+cp "$ADS_DIR/$CONDA_DIR/envs/venv_ads/bin/napari" "$ADS_DIR/$BIN_DIR/ads_napari" || die "Problem creating launchers!"
```

Comment:
Is it best practice to copy a binary from inside the venv like this ? I would think a symbolic link would work just as well without the duplication - but open to either way if this does in fact work

Reply:
Or is this copy required for the Windows compatibility?

Reply:
This is a great question! 

Looking at the history of the original SCT install script, the `cp` lines were added in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/1996 (specifically https://github.com/spinalcordtoolbox/spinalcordtoolbox/commit/213881eb101efad48c7ae39234ee6b94afb6dee1). And, even those changes were replacing an even older script called [`update_bin.sh`](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/91745bbc385d3bb27cb27de247bccf6adb64d970/install/update_bin.sh). Despite this history, though, I can't seem to find any concrete rationale for why we chose to use `cp` over `ln -s`. And, because the decision was made prior to my joining, I don't have any first-hand knowledge, heh.

----

To try and figure out if it's worth it to switch to softlinks, I've spent the past little while doing some reading, trying to understand what possible risks there could be to using a softlink here, but... I couldn't really find any? Softlinks seem perfectly safe and sensible to use here.

That said, I don't think there's too much harm in using a copy, either, as the launcher files are very light files without much actual logic in them besides importing the entry point function. For example, `ads_napari` looks like this:

```python
#!/home/joshua/repos/axondeepseg/ads_conda/envs/venv_ads/bin/python
# -*- coding: utf-8 -*-
import re
import sys
from napari.__main__ import main
if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
    sys.exit(main())
```

Still, that logic comes with some risk? (For example, napari could change its package structure to no longer use [`__main__.py :: main()`](https://github.com/napari/napari/blob/main/napari/__main__.py#L571-L584), along with a user updating ADS using `ads_activate` + `pip install -U`.)

But, because that risk seems relatively small, I'm thinking that it might be better to stick with the "known unknowns" (`cp`) vs. venturing into ["unknown unknowns"](https://en.wikipedia.org/wiki/There_are_unknown_unknowns) (`ln -s`). Maybe there are some risks to softlinks that I've missed (e.g. weird filesystem quirks, network drives, Windows compatibility, etc.) that wouldn't be present with simple copies.

Reply:
>But, because that risk seems relatively small, I'm thinking that it might be better to stick with the "known unknowns" (cp) vs. venturing into ["unknown unknowns"](https://en.wikipedia.org/wiki/There_are_unknown_unknowns) (ln -s). Maybe there are some risks to softlinks that I've missed (e.g. weird filesystem quirks, network drives, weird ComputeCanada restrictions, Windows compatibility, etc.) that wouldn't be present with simple copies.

Ok sounds good! Let's do that then!